superset-frontend/src/filters/components/GroupBy/GroupByFilterPlugin.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ensureIsArray, ExtraFormData, t, tn } from '@superset-ui/core';
+import React, { useEffect, useState } from 'react';
+import { Select } from 'src/common/components';
+import { Styles, StyledSelect } from '../common';
+import { PluginFilterGroupByProps } from './types';
+
+const { Option } = Select;
+
+export default function PluginFilterGroupBy(props: PluginFilterGroupByProps) {
+  const { data, formData, height, width, setDataMask } = props;
+  const { defaultValue, inputRef } = formData;
+
+  const [value, setValue] = useState<string[]>(defaultValue ?? []);
+
+  const handleChange = (value?: string[] | string | null) => {
+    const resultValue: string[] = ensureIsArray<string>(value);
+    setValue(resultValue);
+    const extraFormData: ExtraFormData = {};
+    if (resultValue.length) {
+      extraFormData.interactive_groupby = resultValue;
+    }
+
+    setDataMask({
+      extraFormData,
+    });
```

Comment:
You will also need to set the `filterState` here

Reply:
👍